cmake/legion.cmake

```diff
@@ -16,20 +16,128 @@ if(FF_USE_EXTERNAL_LEGION)
+		  endif()
+		elseif(LINUX_VERSION MATCHES "18.04")
+		  if (CUDA_VERSION VERSION_EQUAL "10.1")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_10.1.243.tar.gz")
+		  elseif (CUDA_VERSION VERSION_EQUAL "10.2")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_10.2.89.tar.gz")
+		  elseif (CUDA_VERSION VERSION_EQUAL "11.0")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.0.3.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.1")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.1.1.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.2")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.2.2.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.3")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.3.1.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.4")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.4.3.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.5")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.5.2.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.6")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.6.2.tar.gz")
+		  elseif(CUDA_VERSION VERSION_EQUAL "11.7")
+		    set(LEGION_URL "https://github.com/flexflow/flexflow-third-party/releases/latest/download/legion_ubuntu-18.04_11.7.0.tar.gz")
+		  endif()
+		endif()
 	endif()
-	if(FF_USE_GASNET)
-	  set(Legion_EMBED_GASNet ON CACHE BOOL "Use embed GASNet")
-	  set(Legion_EMBED_GASNet_VERSION "GASNet-2022.3.0" CACHE STRING "GASNet version")
-	  set(Legion_NETWORKS "gasnetex" CACHE STRING "GASNet conduit")
-	  set(GASNet_CONDUIT ${FF_GASNET_CONDUIT})
+
+	if(LEGION_URL AND NOT FF_USE_GASNET)
+		# Download and import pre-compiled Legion library
+		message(STATUS "Using pre-compiled Legion library")
+		message(STATUS "LEGION_URL: ${LEGION_URL}")
+		set(LEGION_LIBRARY legion)
+		set(REALM_LIBRARY realm)
+		set(LEGION_DOWNLOAD legion)
+
+		set(LEGION_TARBALL_PATH ${CMAKE_BINARY_DIR}/deps/${LEGION_DOWNLOAD}.tar.gz)
+		set(LEGION_EXTRACTED_TARBALL_PATH ${CMAKE_BINARY_DIR}/build/export/${LEGION_DOWNLOAD})
+		set(LEGION_FOLDER_PATH ${CMAKE_BINARY_DIR}/deps/${LEGION_DOWNLOAD})
+		# If LEGION_FOLDER_PATH already exists (this is the case when calling `make install`), don't re-download.
+		if(NOT EXISTS ${LEGION_FOLDER_PATH}/download_succeeded)
+			file(DOWNLOAD ${LEGION_URL} ${LEGION_TARBALL_PATH} STATUS LEGION_DOWNLOAD_RESULT)
```

Comment:
What happens when we update the artifacts in the third-party repo? Do they get automatically re-downloaded, or do the old ones continue to be used?

Reply:
Re-downloaded for each build, or only re-downloaded when the precompiled file changes? If the latter, how is the change detection done?

Reply:
Hmm, okay. At the very least this behavior should be documented somewhere as it seems likely to potentially trip someone (probably me :stuck_out_tongue:) up at some point--that said it can be in a separate docs PR (can you create an issue with these things on there and link it here?)